arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java

```diff
@@ -15,227 +15,358 @@
+import static net.consensys.linea.zktracer.module.ecdata.Trace.TOTAL_SIZE_ECPAIRING_RESULT;
+import static net.consensys.linea.zktracer.module.ecdata.Trace.TOTAL_SIZE_ECRECOVER_DATA;
+import static net.consensys.linea.zktracer.module.ecdata.Trace.TOTAL_SIZE_ECRECOVER_RESULT;
 import static net.consensys.linea.zktracer.types.Containers.repeat;
+import static net.consensys.linea.zktracer.types.Conversions.bigIntegerToBytes;
+import static net.consensys.linea.zktracer.types.Conversions.bytesToUnsignedBytes;
 import static net.consensys.linea.zktracer.types.Utils.leftPadTo;
 
+import java.math.BigInteger;
 import java.util.List;
+import java.util.Optional;
 import java.util.Set;
 
 import com.google.common.base.Preconditions;
+import lombok.Getter;
 import lombok.experimental.Accessors;
-import net.consensys.linea.zktracer.CurveOperations;
 import net.consensys.linea.zktracer.container.ModuleOperation;
 import net.consensys.linea.zktracer.module.ext.Ext;
 import net.consensys.linea.zktracer.module.wcp.Wcp;
 import net.consensys.linea.zktracer.opcode.OpCode;
 import net.consensys.linea.zktracer.types.EWord;
+import net.consensys.linea.zktracer.types.UnsignedByte;
 import org.apache.tuweni.bytes.Bytes;
-import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.crypto.Hash;
+import org.hyperledger.besu.crypto.SECP256K1;
+import org.hyperledger.besu.crypto.SECPPublicKey;
+import org.hyperledger.besu.crypto.SECPSignature;
 
 @Accessors(fluent = true)
 public class EcDataOperation extends ModuleOperation {
-  private static final int EC_RECOVER = 1;
-  private static final int EC_ADD = 6;
-  private static final int EC_MUL = 7;
-  private static final int EC_PAIRING = 8;
-  private static final Set<Integer> EC_TYPES = Set.of(EC_RECOVER, EC_ADD, EC_MUL, EC_PAIRING);
-
-  private static final EWord P =
-      EWord.ofHexString("0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47");
-  private static final EWord Q =
-      EWord.ofHexString("0x30644E72E131A029B85045B68181585D2833E84879B9709143E1F593F0000001");
-  private static final EWord SECP256_K1N =
-      EWord.ofHexString("0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f");
+  private static final Set<Integer> EC_TYPES = Set.of(ECRECOVER, ECADD, ECMUL, ECPAIRING);
+  private static final EWord P_BN = EWord.of(P_BN_HI, P_BN_LO);
+  public static final EWord SECP256K1N = EWord.of(SECP256K1N_HI, SECP256K1N_LO);
 
   private final Wcp wcp;
   private final Ext ext;
 
-  private final int contextNumber;
-  private final int contextNumberDelta;
-  private final Bytes input;
+  @Getter private final int id;
+  @Getter private final int previousId;
+  @Getter private final UnsignedByte[] byteDelta;
+  private final Bytes data;
 
   private final int ecType;
   private final int nRows;
-  /** -1 if no switch off (i.e. preliminary checks passed) */
-  private int hurdleSwitchOffRow;
+  private final int nRowsData;
```

Comment:
This was something from the previous implementation. It can be removed. I am not even sure what was the purpose, but I do not think we need it anymore.